framework/i18n/Formatter.php

```diff
@@ -1092,7 +1092,7 @@ public function asOrdinal($value)
      * If [[sizeFormatBase]] is 1024, [binary prefixes](http://en.wikipedia.org/wiki/Binary_prefix) (e.g. kibibyte/KiB, mebibyte/MiB, ...)
      * are used in the formatting result.
      *
-     * @param integer $value value in bytes to be formatted.
+     * @param string|integer|float $value value in bytes to be formatted.
```

Comment:
HI guys,
probably there is a little misunderstanding, this is my test with i18n Formatter:

PHP
```php
echo 'int 2147483647 => ';
var_dump(Yii::$app->formatter->asSize(2147483647));
echo '<br>';

echo 'int 4147483647 => ';
var_dump(Yii::$app->formatter->asSize(4147483647));
echo '<br>';

echo 'string 2147483647 => ';
var_dump(Yii::$app->formatter->asSize('2147483647'));
echo '<br>';

echo 'string 4147483647 => ';
var_dump(Yii::$app->formatter->asSize('4147483647'));
echo '<br>';

echo '(int)string 2147483647 => ';
var_dump(Yii::$app->formatter->asSize((int)'2147483647'));
echo '<br>';

echo '(int)string 4147483647 => ';
var_dump(Yii::$app->formatter->asSize((int)'4147483647'));
echo '<br>';

echo '(float)string 2147483647 => ';
var_dump(Yii::$app->formatter->asSize((float)'2147483647'));
echo '<br>';

echo '(float)string 4147483647 => ';
var_dump(Yii::$app->formatter->asSize((float)'4147483647'));
echo '<br>';
```

OUTPUT
```
int 2147483647 => string(11) "2 gibibytes" 
int 4147483647 => string(15) "3,863 gibibytes" 
string 2147483647 => string(11) "2 gibibytes" 
string 4147483647 => string(11) "2 gibibytes" 
(int)string 2147483647 => string(11) "2 gibibytes" 
(int)string 4147483647 => string(11) "2 gibibytes" 
(float)string 2147483647 => string(11) "2 gibibytes" 
(float)string 4147483647 => string(15) "3,863 gibibytes" 
```

The problem is the private function formatSizeNumber into yii2/framework/i18n/Formatter.php:

```php
    /**
     * Given the value in bytes formats number part of the human readable form.
     *
     * @param string|integer|float $value value in bytes to be formatted.
     * @param integer $decimals the number of digits after the decimal point
     * @param array $options optional configuration for the number formatter. This parameter will be merged with [[numberFormatterOptions]].
     * @param array $textOptions optional configuration for the number formatter. This parameter will be merged with [[numberFormatterTextOptions]].
     * @return array [parameters for Yii::t containing formatted number, internal position of size unit]
     * @throws InvalidParamException if the input value is not numeric or the formatting failed.
     */
    private function formatSizeNumber($value, $decimals, $options, $textOptions)
    {
        if (is_string($value) && is_numeric($value)) {
            $value = (int) $value;
        }
        if (!is_numeric($value)) {
            throw new InvalidParamException("'$value' is not a numeric value.");
        }
```

The function docs specify input value can is string|integer|float but if $value is a string will cast into integer (also the exposed function asSize have to documented with string|integer|float values).

PHP docs: casting into INTEGER
> The maximum value depends on the system. 32 bit systems have a maximum signed integer range of -2147483648 to 2147483647. So for example on such a system, intval('1000000000000') will return 2147483647. The maximum signed integer value for 64 bit systems is 9223372036854775807.

Fix asSize and asShortSize function docs are only a monition for developer.

The correct way to fix this problem it could be to change:

```php
if (is_string($value) && is_numeric($value)) {
    $value = (int) $value;
} 
```

into:

```php
if (is_string($value) && is_numeric($value)) {
    $value = (float) $value;
} 
```

I hope this is helpful.